server/routes/varyingLicences/handlers/policyChange.ts

```diff
@@ -0,0 +1,189 @@
+      conditionHintText = policyChangeHintText[condition.code][this.formatVersionNumber('2.1')]
+    }
+
+    // Overriding session changeCounter for when the back link is clicked
+    req.session.changedConditionsCounter = conditionCounter
+
+    if (conditionCounter === 0) {
+      return res.redirect(`/licence/vary/id/${licenceId}/policy-changes`)
+    }
+
+    let replacements: AdditionalConditionAp[] | AdditionalConditionPss[] = []
+    if (condition.suggestions) {
+      replacements = await Promise.all(
+        condition.suggestions?.map(async (replacement: { code: string }) =>
+          this.conditionService.getAdditionalConditionByCode(replacement.code)
+        )
+      )
+    }
+
+    switch (condition.changeType) {
+      case conditionChangeType.DELETED:
+        return res.render('pages/vary/policyConditionDeleted', {
+          licenceId,
+          conditionCounter,
+          policyChangesCount,
+          condition,
+          conditionHintText,
+          replacements,
+        })
+      case conditionChangeType.REPLACED:
+        return res.render('pages/vary/policyConditionReplaced', {
+          licenceId,
+          conditionCounter,
+          policyChangesCount,
+          condition,
+          conditionHintText,
+          replacements,
+        })
+      case conditionChangeType.NEW_OPTIONS:
+        return res.render('pages/vary/policyNewOptions', {
+          licenceId,
+          conditionCounter,
+          policyChangesCount,
+          conditionHintText,
+          condition,
+        })
+      // Default to TEXT_CHANGE in case something is wrong with the returned type
+      default:
+        return res.render('pages/vary/policyTextChange', {
+          licenceId,
+          conditionCounter,
+          policyChangesCount,
+          conditionHintText,
+          condition,
+        })
+    }
+  }
+
+  POST = async (req: Request, res: Response): Promise<void> => {
+    const { licence, user } = res.locals
+    const { licenceId, changeCounter } = req.params
+
+    const conditionCounter = +changeCounter
+    const condition = req.session.changedConditions[conditionCounter - 1]
+    const conditionType: LicenceType = await this.conditionService.getAdditionalConditionType(
+      condition.code,
+      (
+        await this.licenceService.getParentLicenceOrSelf(licenceId, user)
+      ).version
+    )
+
+    let additionalLicenceConditions: AdditionalCondition[] = []
+    if (conditionType === LicenceType.AP) {
+      additionalLicenceConditions = licence.additionalLicenceConditions
+    } else if (conditionType === LicenceType.PSS) {
+      additionalLicenceConditions = licence.additionalPssConditions
+    }
+    const licenceCondition = additionalLicenceConditions.find((c: AdditionalCondition) => c.code === condition.code)
+
+    const inputs = []
+    let licenceConditionCodes = additionalLicenceConditions.map((c: AdditionalCondition) => c.code)
+    const changeType =
```

Comment:
load of code here.. presumably we could push some of this down into the API, potentially with new data